scripts/checkMesh/parsing/cli_parsing.py

```diff
@@ -0,0 +1,100 @@
+import argparse
+from collections import OrderedDict
+from dataclasses import dataclass
+import logging
+import sys
+from typing import List, Dict, Set
+
+from . import all_checks_helpers
+
+
+__OPTIONS_SEP = ":"
+__KV_SEP = "="
+
+
+@dataclass(frozen=True)
+class Arguments:
+    vtk_input_file: str
+    checks: OrderedDict()
+
+
+def __get_checks_help_msg() -> str:
+    """
+    Gathers all the doc messages into one string.
+    :return: A string.
+    """
+    tmp = []
+    for check_name, check_helper in all_checks_helpers.items():
+        h = check_name + ": " + check_helper.get_help()
+        tmp.append(h)
+    return "\n".join(tmp)
+
+
+def __parse_cli_options(s: List[str]) -> Dict[str, str]:
+    """
+    Parse a command line option into a dict:
+    :param s: Ths cli string, e.g. "a=1;b=5"
+    :return: The user options as a dict.
+    """
+    result = OrderedDict()
+    for option in s:
+        k, v = option.split(__KV_SEP)
+        result[k] = v
+    return result
+
+
+def validate_cli_options(check_name: str, valid_keys: Set[str], options: Dict[str, str]):
+    """
+    Checks the user input options and logs if an options is not recognized.
+    :param check_name: The key/name of the check.
+    :param valid_keys: The option keys that are valid for the considered check.
+    :param options: The user options.
+    :return: None
+    :todo: Deal with default options.
+    """
+    invalid_keys = set(options.keys()) - valid_keys
+    logging.warning(f"Key(s) [{', '.join(invalid_keys)}] is (are) not valid option(s) of {check_name}. Ignoring.")
+
+
+def parse(cli_args: List[str]) -> Arguments:
+    """
+    Parse the command line arguments and return the corresponding structure.
+    :param cli_args: The list of arguments (as strings)
+    :return: The struct
+    """
+    parser = argparse.ArgumentParser(description='Inspects meshes for GEOSX.')
+    parser.add_argument('-i',
+                        '--vtk-input-file',
+                        metavar='VTK_MESH_FILE',
+                        type=str,
+                        required=True,
+                        dest="vtk_input_file")
+    parser.add_argument('-c',
```

Comment:
Wouldn't it be simpler to have each checker providing a populate parser method which would add option to the parser than passing all through the check option and then parse it?

Reply:
It's possible and it makes sense, but it surely depends on the number of checks will eventually get.
I did not think that way to be honest, but it's a valuable idea.

Reply:
Ok good to me

Reply:
I think I'll got for something like option 2, which looks like a quite efficient way to show the options.
But that pattern somehow induces some kind of introspection to know which options are activated, which are not.
I'll have a look.